From b89a0b11b1133bf2991580203867830747ad4de1 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?S=C3=B8ren=20Sandmann=20Pedersen?= <ssp@localhost.localdomain>
Date: Sun, 21 Aug 2011 10:11:48 -0400
Subject: [PATCH xf86-video-qxl 05/10] Only save the VGA fonts for the primary
 device.

Otherwise, if we try to save the VGA fonts when initializing a
non-primary device, the saving will be routed to the primary one
putting it into VGA mode, which then locks up.
---
 src/qxl_driver.c |    6 ++++--
 1 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/qxl_driver.c b/src/qxl_driver.c
index f3768f7..428735a 100644
--- a/src/qxl_driver.c
+++ b/src/qxl_driver.c
@@ -409,7 +409,8 @@ qxl_save_state(ScrnInfoPtr pScrn)
 {
     qxl_screen_t *qxl = pScrn->driverPrivate;
 
-    vgaHWSaveFonts(pScrn, &qxl->vgaRegs);
+    if (xf86IsPrimaryPci (qxl->pci))
+        vgaHWSaveFonts(pScrn, &qxl->vgaRegs);
 }
 
 static void
@@ -417,7 +418,8 @@ qxl_restore_state(ScrnInfoPtr pScrn)
 {
     qxl_screen_t *qxl = pScrn->driverPrivate;
 
-    vgaHWRestoreFonts(pScrn, &qxl->vgaRegs);
+    if (xf86IsPrimaryPci (qxl->pci))
+        vgaHWRestoreFonts(pScrn, &qxl->vgaRegs);
 }
 #endif /* XSPICE */
 
-- 
1.7.6.2

